src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -2747,73 +2741,73 @@ class GroundingDinoLoss(nn.Module):
     Args:
         matcher (`GroundingDinoHungarianMatcher`):
             Module able to compute a matching between targets and proposals.
-        num_classes (`int`):
-            Number of object categories, omitting the special no-object category.
         focal_alpha (`float`):
             Alpha parameter in focal loss.
         losses (`List[str]`):
             List of all the losses to be applied. See `get_loss` for a list of all available losses.
+        class_reduction (`str`):
+            Specifies the reduction to apply to the label loss. Can be either `'mean'` or `'sum'`
     """
 
-    def __init__(self, matcher, num_classes, focal_alpha, losses):
+    def __init__(self, matcher, focal_alpha, losses, class_reduction):
         super().__init__()
         self.matcher = matcher
-        self.num_classes = num_classes
         self.focal_alpha = focal_alpha
         self.losses = losses
+        self.class_reduction = class_reduction
+
+    def _get_target_classes_one_hot(self, outputs, targets, indices):
+        """
+        Create one_hot based on the matching indices
+        """
+        class_labels = [target["class_labels"] for target in targets]
+        logits = outputs["logits"]
+        label_maps = outputs["label_maps"]
+
+        target_classes_onehot = torch.zeros_like(logits, device=logits.device, dtype=torch.long)
+
+        for i, (source, target) in enumerate(indices):
```

Comment:
What is the type of indices here? This looks like it possibly could be vectorized as well

Reply:
https://github.com/huggingface/transformers/pull/31828/commits/03f6f3ff3cc7620e41d4e8ff9dd5e9eda63e6b4b